src/components/databrowser/hooks/use-fetch-keys.ts

```diff
@@ -0,0 +1,145 @@
+import { useCallback, useRef } from "react"
+import { type SearchFilter } from "@/store"
+import { useRedis } from "@/redis-context"
+import { DATA_TYPES, type DataType } from "@/types"
+import type { Redis } from "@upstash/redis"
+
+const PAGE_SIZE = 30
+
+const FETCH_COUNTS = [100, 200, 400, 800]
+
+export type RedisKey = [string, DataType]
+
+export const useFetchKeys = (search: SearchFilter) => {
```

Comment:
can you write a small docstring here, especially about how the cache works and why we need it?

Reply:
This code is obsolete, I will remove it